sdcm/nemesis.py

```diff
@@ -179,6 +179,8 @@ class Nemesis:  # pylint: disable=too-many-instance-attributes,too-many-public-m
     config_changes: bool = False
     free_tier_set: bool = False     # nemesis should be run in FreeTierNemesisSet
     manager_operation: bool = False  # flag that signals that the nemesis uses scylla manager
+    delete_rows: bool = False  # A flag denotes a nemesis deletes partitions/rows, generating tombstones.
+    NUM_DELETIONS_FACTOR = 5
```

Comment:
Add a comment to explain what it means.

Reply:
Added:
```
+    # NUM_DELETIONS_FACTOR is a constant numeric divisor. It's a factor by which the available-partitions-for-deletion
+    # is divided. This is in order to specify choose_partitions_for_delete(partitions_amount) a reasonable number
+    # of partitions to delete. We prefer not to delete "too many" partitions at once in a single nemesis.
+    # Example usage: partitions_amount=self.tester.partitions_attrs.non_validated_partitions // self.NUM_DELETIONS_FACTOR
```

Reply:
@roydahan Roy, you asked to delete a big number of partitions, when I worked on MV tombstones. Now it will be decreased. Did you change your opinion ?

Reply:
@juliayakovlev , @roydahan , one issue about deleting most of the partitions is, for example, the case of 10M rows-per-partition.
there are currently only 10 such partitions set to be available for deletion. If we delete most/all of them, it'll take a lot of time to rewrite them by s-b and it currently even only has read-stress, so no deletion-nemesis will be able to run afterwards.

Reply:
No, I didn't change my opinion.
The work @juliayakovlev did to have many deletions should remain.

Reply:
@roydahan , it is not "many deletions" rather it is all-available-partitions-for-deletion.
That why `NUM_DELETIONS_FACTOR` was introduced, in order to divide it by 5.
So that more than one nemesis would be able to run and still find partitions for deletion.
If you think it is better, the value of `NUM_DELETIONS_FACTOR` could easily be changed to any other value.

Reply:
So we wanted it to be much higher (was it 100?) and it was intentional.
The next running nemesis should still have rows to delete because there is (should be) a thread that adds rows to these partition ranges (@juliayakovlev please correct me if I'm wrong).

Reply:
That's right - I added s-b commands to prevent that the table will be empty as result of deletions

Reply:
@roydahan , @juliayakovlev , that was right in the past, but not anymore, as in te above mentioned example of `test-cases/longevity/longevity-large-partition-4days.yaml` - it's stress command has only `read`.
Please advise

Reply:
I didn't fully review it, but you cannot undo what Yulia just did.

Reply:
There are 2 large partition longevities that won't run deletion nemeses:
`longevity-large-partition-2days.yaml` and `longevity-large-partition-4days.yaml`

It is due to the `max_partitions_in_test_table` parameter is not defined there
This is the reason why I did not add write command in the stress_cmd part

Reply:
OK, @roydahan , @juliayakovlev , i fixed it so that the existing functionality will not be changed, only the new necessary ones.
Please review this modified code fix:
```
+    def _init_num_deletions_factor(self):
+        # num_deletions_factor is a numeric divisor. It's a factor by which the available-partitions-for-deletion
+        # is divided. This is in order to specify choose_partitions_for_delete(partitions_amount) a reasonable number
+        # of partitions to delete. We prefer not to delete "too many" partitions at once in a single nemesis.
+        # In case 'stress_cmd' has a write-mode and partitions are rewritten, then it is OK to delete all,
+        # (so this factor is set to - 1)
+        # Example usage: partitions_amount=self.tester.partitions_attrs.non_validated_partitions // self.num_deletions_factor
+        self.num_deletions_factor = 5
+        if stress_cmds := self.cluster.params.get('stress_cmd'):
+            if not isinstance(stress_cmds, list):
+                stress_cmds = [stress_cmds]
+            for stress_cmd in stress_cmds:
+                stress_cmd = stress_cmd.split()
+                # In case background load has writes, we can delete all available partitions,
+                # since they are rewritten. Otherwise, we can only delete some of them.
+                if 'scylla-bench' in stress_cmd and '-mode=write' in stress_cmd:
+                    self.num_deletions_factor = 1
+                    break
```

Reply:
ping @roydahan , @juliayakovlev

Reply:
ping-after-2-weeks @roydahan , @juliayakovlev , @fgelcer

Reply:
> ping-after-2-weeks @roydahan , @juliayakovlev , @fgelcer
ping-after-2-weeks @roydahan , @juliayakovlev , @fgelcer